src/Microsoft.Azure.WebJobs/FunctionNameAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// This is for providing a custom name to for an SDK function.
```

Comment:
Rather than introduce our own new attribute for this, I wonder if we shouldn't instead just use DisplayNameAttribute https://msdn.microsoft.com/en-us/library/system.componentmodel.displaynameattribute(v=vs.110).aspx